homework/src/index.js

```diff
@@ -1,47 +1,110 @@
-'use strict';
+"use strick";
+
+function main(url) {
+
+  const root = document.getElementById("root");
 
-{
   function fetchJSON(url, cb) {
     const xhr = new XMLHttpRequest();
     xhr.open('GET', url);
     xhr.responseType = 'json';
     xhr.onload = () => {
-      if (xhr.status < 400) {
+      if (xhr.status === 200) {
         cb(null, xhr.response);
       } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
+        cb({ text: `Error: ${xhr.status} - ${xhr.statusText}`, id: "id", val: "error" });
       }
     };
-    xhr.onerror = () => cb(new Error('Network request failed'));
+    xhr.onerror = () => {
+      cb({ text: "Network request failed", id: "id", val: "error" });
+    };
     xhr.send();
   }
 
-  function createAndAppend(name, parent, options = {}) {
+  function createEl(name, parent, obj) {
     const elem = document.createElement(name);
     parent.appendChild(elem);
-    Object.keys(options).forEach((key) => {
-      const value = options[key];
-      if (key === 'text') {
-        elem.innerText = value;
-      } else {
-        elem.setAttribute(key, value);
-      }
-    });
+    if (obj) {
+      if (obj.text) { elem.innerHTML = obj.text; }
+      if (obj.id) { elem.setAttribute(obj.id, obj.val); }
+    }
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  fetchJSON(url, (err, data) => {
+    if (err === null) {
+      const header = createEl("header", root);
+      createEl("h3", header, { text: "HYF Repositories" });
+      const select = createEl("select", header);
+      let names = [];
+      data.forEach(el => {
+        let name = el.name.charAt(0).toUpperCase() + el.name.slice(1);
```

Comment:
I appreciate that you are trying to make the repository names nicer by making sure that the start with an uppercase letter, but it is better to leave those names untouched because those names are the real names. For instance, there is a `database` repository on the HYF account, but not a `Database` repository.